JWBestPracticeApps/FeedTableViewController/FeedTableViewController/Feed/Playlist.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  Playlist.swift
+//  FeedTableViewController
+//
+//  Created by Amitai Blickstein on 6/26/22.
+//
+
+import Foundation
+import UIKit
+
+/// Setting to `false` will load the landscape videos.
+private let showPortraitVideos = true
+
+struct Playlist {
```

Comment:
Overall, this is a bit confusing. It would be better to not have this and/or refine it to have playlist items created by a builder. Having separate arrays for titles and sources is weird, and makes it difficult to quickly determine what titles are associated with which sources.

It's much better to have an array of JWPlayerItems. For example:
```
var playlist: [JWPlayerItem] = [
     try! JWPlayerItemBuilder().title("Big Big Bunny").file(URL(string: "http://myvideo.url")!).build(),
     try! JWPlayerItemBuilder().title("Paradise").file(URL(string: "http://myvideo.url")!).build(),
     try! JWPlayerItemBuilder().title("Man opens a box of weasels").file(URL(string: "http://myvideo.url")!).build(),
]
```

Of course, these should be valid urls and items so the `try!` works.

Reply:
Removed the intermediary `PlayerItemModel` entirely. Simplified and cleaned up the `Playlist.bpaManual` to just have what the app uses; it should now be clear that `JWPlayerItem` is the true model, `Playlist` is a hard-coded array of models, and the `FeedViewModel` has the view-model logic.